libs/metrics/test_positivity.py

```diff
@@ -216,36 +351,45 @@ def run(
         if input_wide.empty:
             raise NoRealTimeseriesValuesException()
         dates = input_wide.columns.get_level_values(CommonFields.DATE)
-        # The oldest date that is considered recent/not-stale. If recent_days is 1 then this is
-        # the most recent day in the input.
-        assert recent_days >= 1
-        recent_date_cutoff = dates.max() + pd.to_timedelta(1 - recent_days, "D")
+        most_recent_date = dates.max()
 
         methods_with_data = AllMethods._methods_with_columns_available(
             methods, input_wide.index.get_level_values(PdFields.VARIABLE).unique()
         )
         if not methods_with_data:
             raise NoColumnsWithDataException()
 
+        method_map = {method.name: method for method in methods_with_data}
         calculated_dataset_map = {
-            timeseries.DatasetName(method.name): method.calculate(dataset_in, diff_days)
-            for method in methods_with_data
+            method_name: method.calculate(dataset_in, diff_days, most_recent_date)
+            for method_name, method in method_map.items()
         }
         calculated_dataset_recent_map = {
-            name: ds.drop_stale_timeseries(recent_date_cutoff)
-            for name, ds in calculated_dataset_map.items()
+            name: method_output.recent for name, method_output in calculated_dataset_map.items()
         }
+        # HACK: If OldMethod is in the output map (which has Method.recent) then add it again at
+        # the end of the map using Method.all. Remember that dict entries remain in the order
+        # inserted. This makes OldMethod the final fallback for a location if no other Method has
+        # a timeseries for it.
+        old_method_output: Optional[MethodOutput] = calculated_dataset_map.get(
+            timeseries.DatasetName("OldMethod")
+        )
+        if old_method_output is not None:
```

Comment:
you can just do `if old_method_output` right?

Reply:
done